### ISSUE TITLE
Incorrect date result of DateTimeLocalizedStringTransformer

### ISSUE THREAD

**Issue by dkorsak** [symfony/symfony#19721]
https://github.com/symfony/symfony/issues/19531

After merge issue above, there is bug on the DateTimeType::class form type - DateTimeToLocalizedStringTransformer class on branch master.

My timezone is Europe/Warsaw, +2h from UTC timezone.

Part of form:

```php
$builder->add(
                'publishDateEnd',
                DateTimeType::class,
                [
                    'required' => false,
                    'widget' => 'single_text',
                    'format' => 'dd.MM.yyyy HH:mm',
                    'attr' => [
                        'placeholder' => 'dd.mm.yyyy HH:mm',
                    ],
                ]
```

When for example publishDateEnd is set as 2016-08-01 10:00:00 and I set in the web form date as 2016-08-01 12:00:00, after save form date back to 2016-08-01 10:00:00. Should be saved as 2016-08-01 12:00:00.

When I add 'view_timezone' => 'UTC' into widget option, date is display as correct date, but is invalid in database

My php.ini timezone is set as Europe/Warsaw

**Comment by Jean85** [comment]
Thanks for the bug report!

Does this but happen with all possible inputs, while using your timezone? 
What about other timezones (like Europe\Rome, the one referenced in #19531)?
Can you write a test that reproduces the issue?

**Comment by dkorsak** [comment]
Bug is for every timezone.  if i will not set timezone, in this case defaut timezone is used from php.ini.

**Comment by sbarbo** [comment]
Hi, i've got the same pb when using bootstrap datetimepicker (with sonata admin). I fixed it setting UTC as timezone for my projets, as my datetime as stored witout time zone in database, it is not a big issue. A way to correct it is to set the timezone, even if inputTimezone and outputTimezone are the same.

**Comment by Jean85** [comment]
So you think that a fix would be setting the default timezone, @sbarbo ?

**Comment by sbarbo** [comment]
Ok, i think i found the issue.
In the reverseTransform function, we call that
            $dateTime = new \DateTime(sprintf('@%s', $timestamp), new \DateTimeZone($this->outputTimezone));

But we set a timestamp in the constructor, so, the timezone is ignore (as say php documentation).
If we just add 
            $dateTime->setTimezone(new \DateTimeZone($this->outputTimezone));
after the construction of the datetime, it's work well.

the new code should be 
```
$dateTime = new \DateTime(sprintf('@%s', $timestamp));
$dateTime->setTimeZone(new \DateTimeZone($this->outputTimezone));
```

**Comment by mbeccati** [comment]
Thanks @sbarbo

**Comment by Jean85** [comment]
Thank you @mbeccati!

Status: reviewed

**Comment by sbarbo** [comment]
Thanks @mbeccati . I don't know how the fix are merged, but the issue is also in the 2.8 version.

**Comment by Jean85** [comment]
The PR is on the 2.7 branch, it will be ported on newer version after merging.
